File: ios/Samples/Showcase-ObjectiveC/Showcase-ObjectiveC/ViewControllers/Others/WebScreenletViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
@import LiferayScreens;
#import "LiferayLogger.h"
#import "WebScreenletViewController.h"

@interface WebScreenletViewController () <WebScreenletDelegate>

@property (weak, nonatomic) IBOutlet WebScreenlet *screenlet;

@end

@implementation WebScreenletViewController

- (void)viewDidLoad {
    [super viewDidLoad];
	
	self.screenlet.delegate = self;

	NSString *url = [LiferayServerContext stringPropertyForKey:@"webUrl"];
	
	WebScreenletConfigurationBuilder *builder = [[WebScreenletConfigurationBuilder alloc] initWithUrl:url];
	
	[builder setWithWebType:WebTypeLiferayAuthenticated];
	[builder addJsWithLocalFile:@"gallery"];
	[builder addCssWithLocalFile:@"gallery"];
	[builder addCssWithLocalFile:@"bigger_pagination"];
	
	WebScreenletConfiguration *configuration = [builder load];
	
	self.screenlet.configuration = configuration;
	[self.screenlet load];
}

- (void)onWebLoad:(WebScreenlet *)screenlet url:(NSString *)url {
	LiferayLog(url)
}

- (void)screenlet:(WebScreenlet *)screenlet onError:(NSError *)error {
	LiferayLog(error.debugDescription);
}

- (void)screenlet:(WebScreenlet *)screenlet onScriptMessageNamespace:(NSString *)namespace_ onScriptMessage:(NSString *)message {
	LiferayLog(namespace_)
	LiferayLog(message)
}

@end

File: ios/Samples/Showcase-swift/LiferayScreens-Showcase-Swift/ViewControllers/Portlet/WebScreenletViewController.swift
/**
* Copyright (c) 2000-present Liferay, Inc. All rights reserved.
*
* This library is free software; you can redistribute it and/or modify it under
* the terms of the GNU Lesser General Public License as published by the Free
* Software Foundation; either version 2.1 of the License, or (at your option)
* any later version.
*
* This library is distributed in the hope that it will be useful, but WITHOUT
* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
* FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
* details.
*/
import UIKit
import LiferayScreens

class WebScreenletViewController: UIViewController, WebScreenletDelegate {

	// MARK: Outlets

	@IBOutlet var screenlet: WebScreenlet? {
		didSet {
			screenlet?.delegate = self
			screenlet?.presentingViewController = self
		}
	}

	// MARK: UIViewController

	override func viewDidLoad() {
		super.viewDidLoad()

		let url = "/web/guest"

		let configuration = WebScreenletConfigurationBuilder(url: url)
				.load()

		screenlet?.configuration = configuration

		screenlet?.load()
	}

	// MARK: WebScreenletDelegate

    func onWebLoad(_ screenlet: WebScreenlet, url: String) {

		
    }

	func screenlet(_ screenlet: WebScreenlet, onError error: NSError) {
		LiferayLogger.logDelegateMessage(args: error)
	}

	func screenlet(_ screenlet: WebScreenlet, onScriptMessageNamespace namespace: String, onScriptMessage message: String) {

		//We can check what is the name of the message handler responsible for the action
		performSegue(withIdentifier: "detail", sender: message)
	}

	override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
		let vc = segue.destination as? WebDetailViewController
		vc?.url = "http://" + (sender as! String)
	}
}
